lib/openshift/cluster_version.rb

```diff
@@ -6,14 +6,63 @@ def channel(user: nil, cached: true, quiet: false)
       rr.dig('spec', 'channel')
     end
 
-    def upstream(user: nil, cached: true, quiet: false)
+    def upstream(user: nil, cached: false, quiet: false)
       rr = raw_resource(user: user, cached: cached, quiet: quiet)
       rr.dig('spec', 'upstream')
     end
 
-    def version(user: nil, cached: true, quiet: false)
+    def version(user: nil, cached: false, quiet: false)
```

Comment:
here and a couple of mroe places where still `cached: false`

Reply:
@akostadinov, I've updated the default to true for all calls except `upgrade_completed` and `wait_for_upgrade_completion` since I've been bitten when the `cache: true` for default for those two methods.  PTAL and merge if there are no more issues.

Reply:
`wait_for_upgrade_completion` should **not** have the `cached` option at all. Wait method should call `upgrade_completed(..., cached: false)`. And `upgrade_completed` should default to `cached: true`. Generic workflow is that you *getter* methods will returned cached values when available. So if you call multiple methods you do not perform multiple calls to API. Point in time checks. When you want updated data, you need to call a getter method with `cached: false`.

Reply:
@akostadinov, I get it now... ok, PR updated, PTAL.  Thanks!

Reply:
@akostadinov similarly, should `cached` set to true for this method? https://github.com/openshift/verification-tests/blob/master/lib/openshift/resource.rb#L282